core/Pos/Core/Types.hs

```diff
@@ -283,6 +285,7 @@ instance Bounded Coin where
 maxCoinVal :: Word64
 maxCoinVal = 45000000000000000
 
+-- FIXME: This operation is unsafe because it doesn't check 'maxCoinVal'.
```

Comment:
Funny thing is that `getCoin` has `unsafe` prefix despite being safe, while `mkCoin` is unsafe and doesn't have it :)